### ISSUE TITLE
Show "searching..." + loading spinner when using `refresh`

### ISSUE THREAD

**Issue by nvquanghuy** [angular-ui/ui-select#919]
I know that I can use `refresh` to asynchornously fetch data from API server. Is there a way for me to give some visual indication to the user that we're loading the data? Similar to how select2 has a spinner, and a "Searching..." text: http://i.imgur.com/3BsemlD.png

http://select2.github.io/select2/ (please see the Loading Remote Data example)

Or if this is not currently supported, I'm happy to open the code and contribute a patch, I just need to be pointed at a right direction.

Thanks!

**Comment by moneytree-doug** [comment]
This is a nice feature that I would like too. My use case is that my list is just too long and it takes a little time to load.

**Comment by bmherold** [comment]
+1

**Comment by think2011** [comment]
+1

**Comment by jarool** [comment]
+1

**Comment by ggondim** [comment]
++1

**Comment by sanderaido** [comment]
+1

**Comment by venkatesh13** [comment]
+1

**Comment by rusboy** [comment]
+1

**Comment by sebaas** [comment]
+1

**Comment by Rykie** [comment]
+2

**Comment by thiagoross** [comment]
+1

**Comment by ErikDubbelboer** [comment]
I think this is something most people like to have implemented differently. To accommodate for everyones wishes would complicate the code quite a bit (as you can already see with other features). It's not easy for people to currently implement this themselves so I don't think it's good idea to put this into the library.

I think it's better if there would be some good documentation on the various ways people can implement this themselves.

We use `ui-disable-choice="item.id < 0"` and in our 'refresh(val)' function we set `$scope.list = [{id: -1, name:'Searching for ' + val}];`.

We also have an `ng-class="{'loading': loading}"` on the `ui-select` and during the ajax call we set `$scope.loading = true`. Then we use some css to add a loading indicator:
```css
.loading:after {
z-index: 10;
    border: solid rgba(9, 62, 78, .2);
    border-top-color: #189dc8;
    -webkit-border-radius: 50%;
    -moz-border-radius: 50%;
    border-radius: 50%;
    -webkit-animation: spin 1s infinite linear;
    -moz-animation: spin 1s infinite linear;
    animation: spin 1s infinite linear;
    content: "";
    position: absolute;
    left: 50%;
    top: 50%;
    width: 40px;
    height: 40px;
    margin: -20px 0 0 -20px;
    border-width: 5px;
}
```

**Comment by jihunleekr** [comment]
+1

**Comment by julianosaless** [comment]
+30

**Comment by rafa-aguilar** [comment]
+1

**Comment by shprink** [comment]
+1

**Comment by dimylik** [comment]
+1

**Comment by peirick** [comment]
+1

**Comment by kiddo13** [comment]
#1780 Display loading text on refresh

**Comment by kiddo13** [comment]
I try to create a loading text that would be displayed if refreshAttr return $http promise.
Loading text could be styled with css class .ui-select-refreshing, which you can use to add spinner.

**Comment by user378230** [comment]
Closed by #1779
